src/main/java/name/webdizz/sonar/grammar/sensor/GrammarIssuesWrapper.java

```diff
@@ -0,0 +1,225 @@
+package name.webdizz.sonar.grammar.sensor;
+
+import static com.google.common.base.Preconditions.checkState;
+import java.util.Objects;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.component.ResourcePerspectives;
+import org.sonar.api.issue.Issuable;
+import org.sonar.api.issue.Issue;
+import org.sonar.api.rule.RuleKey;
+
+/**
+ * Wrapper for one code line to check
+ *
+ * @author Oleg_Sopilnyak1
+ */
+public class GrammarIssuesWrapper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GrammarIssuesWrapper.class);
+
+    private String line;
+    private int lineNumber;
+    private ResourcePerspectives perspectives;
+    private InputFile inputFile;
+    private RuleKey ruleKey;
+
+    private GrammarIssuesWrapper() {
+    }
+
+    public String getLine() {
+        return line;
+    }
+
+    public int getLineNumber() {
+        return lineNumber;
+    }
+
+    public String getKey(){
+        return inputFile.absolutePath();
+    }
+    /**
+     * Something wrong with wrapped line detected.<BR/>
+     * Appropriate issue will be created.
+     *
+     * @param message Incident's explanation
+     * @param column column number in wrapped line
+     */
+    public void incident(final String message, final int column) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Reported about incident \"{}\" Col. {}", message, column);
+        }
+        final Issuable issuable = perspectives.as(Issuable.class, inputFile);
+        final Issue issue = issuable.newIssueBuilder()
+                .message(message)
+                .line(new Integer(lineNumber))
+                .ruleKey(ruleKey)
+                .attribute("Col.", Integer.toString(column))
+                .build();
+        final boolean success = issuable.addIssue(issue);
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Issue about incident added {}", success);
+        }
+    }
+
+    /**
+     * Something wrong with wrapped line detected.<BR/>
+     * Appropriate issue will be created.
+     *
+     * @param message Incident's explanation
+     */
+    public void incident(final String message) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Reported about incident \"{}\"", message);
+        }
+        final Issuable issuable = perspectives.as(Issuable.class, inputFile);
+        final Issue issue = issuable.newIssueBuilder()
+                .message(message)
+                .line(new Integer(lineNumber))
+                .ruleKey(ruleKey)
+                .build();
+        final boolean success = issuable.addIssue(issue);
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Issue about incident has added {}", success);
+        }
+    }
+
+    @Override
+    public int hashCode() {

```

Comment:
Object methods are better to be placed at the class end as there is no such interest in them